include/ethercat_sdk_master/EthercatActuatorDevice.h

```diff
@@ -0,0 +1,73 @@
+/*
+ ** Copyright 2020 Robotic Systems Lab - ETH Zurich:
+ ** Lennart Nachtigall, Jonas Junger
+ ** Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+ **
+ ** 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+ **
+ ** 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+ **
+ ** 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
+ **
+ ** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "EthercatDevice.hpp"
+
+namespace ecat_master {
+class EthercatActuatorDevice: public EthercatDevice
+{
+
+public:
+    typedef std::shared_ptr<EthercatActuatorDevice> SharedPtr;
```

Comment:
`using SharedPtr = std::shared_ptr<EthercatActuatorDevice>;`

Reply:
https://stackoverflow.com/questions/10747810/what-is-the-difference-between-typedef-and-using-in-c11
Given this answer I would guess a typedef is the way to go?

Reply:
No, actually `using` is preferred because it allows alias templates: 

```cpp
template<typename ValueType>
using my_special_map = std::map<key_t, ValueType>;
```

Also the syntax is more natural compared to assignments: `Something = Expression`

`typedefs` are allowed in a few more contexts but these cases are very exotic. (You can use typedef in `if()` and `for()` statements).

Reply:
Reference to an author with more authority: https://google.github.io/styleguide/cppguide.html#Aliases